moon-store-mac-os/moon-store-mac-os/Domain/Repositories/AuthenticationRepository.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  LoginRepository.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/14/24.
+//
+
+import Foundation
+
+final class AuthenticationRepository: BaseNetworkService {
+    var loggedUser: UserModel? {
+        guard let storedUser = try? store.fetch().first else { return nil }
+        return UserModel.factory(from: storedUser)
+    }
+    
+    var isLoggedUser: Bool { loggedUser != nil }
+    
+    private let decoder: JSONDecoder = .init()
+    private let store: DataManager<UserSwiftDataModel> = .init()
+    
+    // MARK: - Methods
+    
+    func login(email: String,
+               password: String) async throws {
+        guard !isLoggedUser else { return }
+        
+        let parameters: [String: Any] = [
+            "email": email,
+            "password": password
+        ]
+        
+        do {
+            let response = try await postData(for: MSEndpoint.login.path,
+                                              with: parameters)
+            let loginResponse = try decoder.decode(LoginResponse.self, from: response)
+            try storeUser(loginResponse.data)
+        } catch let error as MSError {
+            throw error
+        }
```

Comment:
Silent error when the `let error` is not `MSError`, you should add a default catch to handle the general errors.

Reply:
I always return a MSError from hold